src/gpl/src/mbff.cpp

```diff
@@ -43,58 +43,385 @@
 #include <chrono>
 #include <ctime>
 #include <iostream>
-#include <map>
 #include <memory>
 #include <random>
-#include <set>
-#include <vector>
-
-/*
-height and width of a flip-flop
-ratios are used for:
-    (1) finding all slot locations in a tray
-    (2) finding power consumption per slot
-*/
 
 namespace gpl {
 
-constexpr int NUM_SIZES = 7;
-constexpr float WIDTH = 2.448, HEIGHT = 1.2;
-constexpr float RATIOS[NUM_SIZES]
-    = {1, 0.95, 0.875, 0.854, 0.854, 0.844, 0.844};
+// check if flops with 1 Q pin are inverting
+bool MBFF::IsInverting(odb::dbInst* inst)
+{
+  sta::Cell* cell = network_->dbToSta(inst->getMaster());
+  if (cell == nullptr) {
+    return false;
+  }
+  sta::LibertyCell* lib_cell = network_->libertyCell(cell);
+  if (lib_cell == nullptr) {
+    return false;
+  }
+  return lib_cell->isInverter();
```

Comment:
This checks if the cell is an inverter which no FF is.  It doesn't not tell you if the edge is inverting.